rust/src/data_catalog/unity/mod.rs

```diff
@@ -0,0 +1,132 @@
+//! Databricks Unity Catalog.
+//!
+//! This module is gated behind the "unity" feature.
+use super::{DataCatalog, DataCatalogError};
+use reqwest::header;
+use reqwest_middleware::{ClientBuilder, ClientWithMiddleware};
+use reqwest_retry::{policies::ExponentialBackoff, RetryTransientMiddleware};
+use serde::Deserialize;
+
+/// Databricks Unity Catalog - implementation of the `DataCatalog` trait
+pub struct UnityCatalog {
+    client_with_retry: ClientWithMiddleware,
+    workspace_url: String,
+}
+
+impl UnityCatalog {
+    /// Creates a new UnityCatalog
+    pub fn new() -> Result<Self, DataCatalogError> {
+        let token_var = "DATABRICKS_ACCESS_TOKEN";
+        let access_token =
+            std::env::var(token_var).map_err(|_| DataCatalogError::MissingEnvVar {
+                var_name: token_var.into(),
+            })?;
+
+        let auth_header_val = header::HeaderValue::from_str(&format!("Bearer {}", &access_token))
+            .map_err(|_| DataCatalogError::InvalidAccessToken)?;
+
+        let headers = header::HeaderMap::from_iter([(header::AUTHORIZATION, auth_header_val)]);
+        let client = reqwest::Client::builder()
+            .default_headers(headers)
+            .build()?;
+
+        let retry_policy = ExponentialBackoff::builder().build_with_max_retries(10);
+
+        let client_with_retry = ClientBuilder::new(client)
+            .with(RetryTransientMiddleware::new_with_policy(retry_policy))
+            .build();
+
+        let workspace_var = "DATABRICKS_WORKSPACE_URL";
+        let workspace_url =
+            std::env::var(workspace_var).map_err(|_| DataCatalogError::MissingEnvVar {
+                var_name: workspace_var.into(),
+            })?;
+
+        Ok(Self {
+            client_with_retry,
+            workspace_url,
+        })
+    }
+}
+
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum TableResponse {
+    Success { storage_location: String },
+    Error { error_code: String, message: String },
+}
+
+/// Possible errors from the unity-catalog/tables API call
+#[derive(thiserror::Error, Debug)]
+pub enum GetTableError {
```

Comment:
Are we using these error variants somewhere?

While still very much a long term things, we have been trying to reduce the number of error variants we expose to users in this crate and also make them more actionable. Maybe we can make some common errors explicit - like 403?

Reply:
I did it in a similar way to the existing AWS glue module but I understand it's not perfect.

Didn't want to start refactoring existing code either. :D At least not yet.